codemodder/codemods/harden_ruamel.py

```diff
@@ -1,52 +1,38 @@
 import libcst as cst
-from libcst.codemod import CodemodContext
 from libcst import matchers
-from codemodder.codemods.base_codemod import (
-    SemgrepCodemod,
-    CodemodMetadata,
-    ReviewGuidance,
-)
-from codemodder.codemods.base_visitor import BaseTransformer
-from codemodder.codemods.change import Change
-from codemodder.file_context import FileContext
+from codemodder.codemods.base_codemod import ReviewGuidance
+from codemodder.codemods.api import SemgrepCodemod
 
 
-class HardenRuamel(SemgrepCodemod, BaseTransformer):
-    METADATA = CodemodMetadata(
-        DESCRIPTION=("Ensures all unsafe calls to ruamel.yaml.YAML use `typ='safe'`."),
-        NAME="harden-ruamel",
-        REVIEW_GUIDANCE=ReviewGuidance.MERGE_WITHOUT_REVIEW,
-    )
-    CHANGE_DESCRIPTION = METADATA.DESCRIPTION
-    YAML_FILES = [
-        "harden-ruamel.yaml",
-    ]
+class HardenRuamel(SemgrepCodemod):
+    NAME = "harden-ruamel"
+    REVIEW_GUIDANCE = ReviewGuidance.MERGE_WITHOUT_REVIEW
+    DESCRIPTION = "Ensures all unsafe calls to ruamel.yaml.YAML use `typ='safe'`."
 
-    def __init__(self, codemod_context: CodemodContext, file_context: FileContext):
-        SemgrepCodemod.__init__(self, file_context)
-        BaseTransformer.__init__(
-            self,
-            codemod_context,
-            self._results,
-            file_context.line_exclude,
-            file_context.line_include,
-        )
+    @classmethod
+    def rule(cls):
+        return """
+            rules:
+                - pattern-either:
+                  - patterns:
+                    - pattern: ruamel.yaml.YAML(typ="unsafe", ...)
+                    - pattern-inside: |
+                        import ruamel
+                        ...
+                  - patterns:
+                    - pattern: ruamel.yaml.YAML(typ="base", ...)
+                    - pattern-inside: |
+                        import ruamel
+                        ...
 
-    def leave_Call(self, original_node: cst.Call, updated_node: cst.Call):
-        pos_to_match = self.node_position(original_node)
-        if self.filter_by_result(
-            pos_to_match
-        ) and self.filter_by_path_includes_or_excludes(pos_to_match):
-            line_number = pos_to_match.start.line
-            self.CHANGES_IN_FILE.append(
-                Change(str(line_number), self.CHANGE_DESCRIPTION).to_json()
-            )
-            new_args = update_arg_target(original_node.args, target_arg="typ")
-            return updated_node.with_changes(args=new_args)
-        return updated_node
+        """
+
+    def on_result_found(self, original_node, updated_node):
+        new_args = make_new_args(original_node.args, target_arg="typ")
+        return self.update_arg_target(updated_node, new_args)
 
 
-def update_arg_target(original_args, target_arg):
+def make_new_args(original_args, target_arg):
```

Comment:
Is this more generally useful now? Could it move to the `Helpers` class?

Reply:
good suggestion, did that